controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -65,47 +97,460 @@ func (r *StorageClassClaimReconciler) Reconcile(ctx context.Context, request rec
+				},
+			},
+		}
+		return nil
+	})
+
+	if err != nil {
+		r.log.Error(
+			err,
+			"Failed to update CephBlockPool.",
+			"CephBlockPool",
+			klog.KRef(r.cephBlockPool.Namespace, r.cephBlockPool.Name),
+		)
+		return err
+	}
+
+	cephClients := map[string]string{
+		"provisioner": r.cephClientProvisioner.Name,
+		"node":        r.cephClientNode.Name,
+	}
+	phase := ""
+	if r.cephBlockPool.Status != nil {
+		phase = string(r.cephBlockPool.Status.Phase)
+	}
+
+	r.setCephResourceStatus(r.cephBlockPool.Name, "CephBlockPool", phase, cephClients)
+
+	r.storageConsumer.Status.GrantedCapacity = r.storageConsumer.Spec.Capacity
+
+	return nil
+}
+
+func (r *StorageClassClaimReconciler) reconcileCephFilesystemSubVolumeGroup() error {
+
+	cephFilesystemList := rookCephv1.CephFilesystemList{}
+	if err := r.Client.List(r.ctx, &cephFilesystemList, client.InNamespace(r.namespace)); err != nil {
+		return fmt.Errorf("Error fetching CephFilesystemList. %+v", err)
+	}
+
+	var cephFileSystemName string
+	availableCephFileSystems := len(cephFilesystemList.Items)
+	if availableCephFileSystems == 0 {
+		return fmt.Errorf("No CephFileSystem found in the cluster")
+	}
+	if availableCephFileSystems > 1 {
+		klog.Warningf("More than one CephFileSystem found in the cluster, selecting the first one")
+	}
+
+	cephFileSystemName = cephFilesystemList.Items[0].Name
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.cephFilesystemSubVolumeGroup, func() error {
+		if err := r.own(r.cephFilesystemSubVolumeGroup); err != nil {
+			return err
+		}
+
+		r.cephFilesystemSubVolumeGroup.Spec = rookCephv1.CephFilesystemSubVolumeGroupSpec{
+			FilesystemName: cephFileSystemName,
+		}
+		return nil
+	})
+
+	if err != nil {
+		r.log.Error(
+			err,
+			"Failed to update CephFilesystemSubVolumeGroup.",
+			"CephFilesystemSubVolumeGroup",
+			klog.KRef(r.cephFilesystemSubVolumeGroup.Namespace, r.cephFilesystemSubVolumeGroup.Name),
+		)
+		return err
+	}
+
+	cephClients := map[string]string{
+		"provisioner": r.cephClientProvisioner.Name,
+		"node":        r.cephClientNode.Name,
+	}
+	phase := ""
+	if r.cephFilesystemSubVolumeGroup.Status != nil {
+		phase = string(r.cephFilesystemSubVolumeGroup.Status.Phase)
+	}
+
+	r.setCephResourceStatus(r.cephFilesystemSubVolumeGroup.Name, "CephFilesystemSubVolumeGroup", phase, cephClients)
+
+	return nil
+}
+
+func (r *StorageClassClaimReconciler) reconcileCephClientRBDProvisioner() error {
+
+	desired := &rookCephv1.CephClient{
+		Spec: rookCephv1.ClientSpec{
+			Caps: map[string]string{
+				"mon": "profile rbd",
+				"mgr": "allow rw",
+				"osd": fmt.Sprintf("profile rbd pool=%s", r.cephBlockPool.Name),
+			},
+		},
+	}
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.cephClientProvisioner, func() error {
+		if err := r.own(r.cephClientProvisioner); err != nil {
+			return err
+		}
+
+		addStorageRelatedAnnotations(r.cephClientProvisioner, r.storageClassClaim.Name, "rbd", "provisioner")
+		r.cephClientProvisioner.Spec = desired.Spec
```

Comment:
Why not embed the desired state here like we did on all the other reconcile functions? (and remove it from line 353 - 361)
```suggestion
		r.cephClientProvisioner.Spec = rookCephv1.ClientSpec{
			Caps: map[string]string{
				"mon": "profile rbd",
				"mgr": "allow rw",
				"osd": fmt.Sprintf("profile rbd pool=%s", r.cephBlockPool.Name),
			},
		}
```

Reply:
ack